src/GithubFilesBundle/Controller/DefaultController.php

```diff
@@ -46,16 +50,33 @@ public function getFilesAction(Request $request){
 
         $owner = $request->query->get('owner');
         $repoName = $request->query->get('repo');
-        $path = $request->query->get('path');
+        $sha = $request->query->get('sha');
+
+        if( empty($sha)){
+            $url = "/repos/:owner/:repo/contents/";
+        }else{
+            $url = "/repos/:owner/:repo/git/blobs/:sha";
+        }
 
-        $listFiles = $api->get('/repos/:owner/:repo/contents/:path', [
+        $blob = $api->get($url, [
             'owner' => $owner,
             'repo' => $repoName,
-            'path' => $path
+            'sha' => $sha
         ]);
 
         return $response->setData(array(
-            'data' => $api->decode($listFiles)
+            'data' => $api->decode($blob)
         ));
     }
+
+    public static function parseLink($link, $rel){
```

Comment:
Why is this a static function?  Should it be moved to a Util class instead of the controller?